sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs

```diff
@@ -13,11 +13,13 @@
 namespace Azure.Messaging.EventHubs.Processor
 {
     /// <summary>
-    ///   Constantly receives <see cref="EventData" /> from every partition in the context of a given consumer group.
-    ///   The received data is sent to an <see cref="IPartitionProcessor" /> to be processed.
+    ///   Constantly receives <see cref="EventData" /> from partitions in the context of a given consumer group.  The received
+    ///   data is sent to a partition processor to be processed.
     /// </summary>
     ///
-    public class EventProcessor
+    /// <typeparam name="T">The type of partition processor used by this instance by default; the type must be a <see cref="BasePartitionProcessor" /> and must have a parameterless constructor.</typeparam>
+    ///
+    public class EventProcessor<T> where T : BasePartitionProcessor, new()
     {
```

Comment:
The only `private` property I think we could set to `protected` is the `InnerClient`, in case we delegate the client creation to the `EventProcessor` in the future. We should keep it `private` if we keep things as they are.

Reply:
Better private and have to adjust than the other way around.   👍